search/load_data.py

```diff
@@ -0,0 +1,162 @@
+import re
+
+import pandas as pd
+from psycopg2.extras import NumericRange
+
+from search.models import (
+    Address,
+    ContactData,
+    School,
+    HighSchoolClass,
+    SubjectName,
+    ExtendedSubject,
+    LanguageName,
+    Language,
+    Statistics,
+)
+
+
+def get_class_type(type: str) -> str:
+    class_type = {
+        "ogólnodostępny": "O",
+        "mistrzostwa sportowego": "MS",
+        "dwujęzyczny": "D",
+        "międzynarodowy": "M",
+        "wstępny": "DW",
+        "sportowy": "S",
+        "integracyjny cz. ogólnodostępna": "I-o",
+        "integracyjny cz. dla kandydatów z orzeczeniem o potrzebie kształcenia specjalnego": "I-i",
+        "przygotowania wojskowego": "PW",
+    }
+    return class_type.get(type, "")
+
+
+def insert_publiczne_oddzialy_2020() -> None:
+    data = pd.read_csv("csvs/publiczne_oddzialy_2020.csv", sep=";").to_dict(orient="records")
+    for item in data:
+        regon = str(item.get("REGON szkoły")).zfill(9)
+        try:
+            # get existing school object
+            school = School.objects.get(public_institution_data__regon=regon)
+        except School.DoesNotExist:
+            # add new school
+            address = Address.objects.create(
+                city=item.get("Miejscowość szkoły"),
+                postcode=item.get("Kod pocztowy szkoły"),
+                district=item.get("Dzielnica"),
+                street=item.get("Ulica szkoły"),
+                building_nr=item.get("Nr budynku szkoły"),
+                longitude=item.get(""),  # no data!
+                latitude=item.get(""),  # no data!
+            )
+            contact = ContactData.objects.create(
+                website=item.get(""),  # no data!
+                phone=item.get(""),  # no data!
+                email=item.get(""),  # no data!
+            )
+            # public_institution_data = PublicInstitutionData.objects.create(
+            #     short_name=item.get(""),  # no data!
+            #     institution_name=item.get(""),  # no data!
+            #     institution_short_name=item.get(""),  # no data!
+            #     institution_type=item.get(""),  # no data!
+            #     institution_nr=item.get(""),  # no data!
+            #     institution_RSPO=item.get(""),  # no data!
+            #     RSPO=item.get(""),  # no data!
+            #     institution_regon=item.get(""),  # no data!
+            #     regon=item.get(""),  # no data!
+            #     data=item.get(""),  # no data!
+            # )
+            school = School.objects.create(
+                school_name=item.get("Nazwa szkoły"),
+                school_type=item.get("Typ szkoły (LO/LP/TECH/BR1St/LU/TU/PS)"),
```

Comment:
This should already be specified in code a long time ago, but the school_type can be:
przedszkole, liceum ogólnokształcące, technikum, szkoła podstawowa, szkoła policealna. 
School type generalized can be:
przedszkole, szkoła ponadpodstawowa, szkoła policealna, szkoła podstawowa

Reply:
After #34 the school_type has choices set and school_type_generlized is removed